### PR TITLE
Setup webhook

### PR DESCRIPTION
**Pull request by mlsaito**
PR for Issue #2 

* Remove unnecessary third-party components (middleware, etc.)
* Setup webhook for local env.
* Update README.md

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,6 +26,8 @@ vendor
 logs
 *.log
 *DS_Store
+*.env
```
Comment (kimmy) [inline .gitignore]: Actually, I think it's okay not to include `.env` to `.gitignore` so collaborators or maintainers will have an idea as to what env variables are necessary. Each person could then have a `.env.development` or `.env.test` locally on their own machines, and these are the files that will be included in `.gitignore`

File: controllers/controllers.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package controllers
+
+import (
+  // "encoding/json"
```
Comment (kimmy) [inline controllers/controllers.go]: is this necessary?

File: controllers/controllers.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+func TelegramUpdate(w http.ResponseWriter, r *http.Request) {
+  log := logger.GetLogger()
+  
+  bodyBytes, _ := ioutil.ReadAll(r.Body)
+  bodyString := string(bodyBytes)
+
+  log.Debugf("Request Body: %s", bodyString)
+
+  // Write to db here
```
Comment (kimmy) [inline controllers/controllers.go]: Just a question: this really writes to db?

File: controllers/controllers.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  w.WriteHeader(http.StatusOK)
+}
+
+// // Post is used to create a new user object
```
Comment (kimmy) [inline controllers/controllers.go]: Can we remove these comments? line `28` to `51`

File: logger/logger.go [hunk 5]
```diff
@@ -1,30 +1,30 @@
 package logger
 
 import (
-	"os"
+  "os"
```
Comment (kimmy) [inline logger/logger.go]: Sorry, I think we need to fix spacing on this file. `We use tabs for indentation and gofmt emits them by default. Use spaces only if you must.`

File: main.go [hunk 6]
```diff
@@ -1,31 +1,79 @@
...
 
-	port := defaults.String(os.Getenv("GO_ENV_PORT"), "8000")
-
-	log.Debugf("Starting API server at %s", port)
 	log.Debugf("TOKEN: %s", os.Getenv("TOKEN"))
 
-	//Start the api here
-	http.ListenAndServe(fmt.Sprintf(":%s", port), app)
+
+	// Initializing Telegram Bot . . .
```
Comment (kimmy) [inline main.go]: I think it's safe to remove this comment?

File: main.go [hunk 7]
```diff
@@ -1,31 +1,79 @@
...
+	bot, err := tgbotapi.NewBotAPI(os.Getenv("TOKEN"))
+	if err != nil {
+		log.Fatal("Error initializing tgbotapi")
+	}
+
+	bot.Debug = true
+	log.Debugf("Authorized on account %s", bot.Self.UserName)
+
+  // ngrok for local development
```
Comment (kimmy) [inline main.go]: if comments are necessary, indent them properly?

File: main.go [hunk 8]
```diff
@@ -1,31 +1,79 @@
...
+  // Run ngrok on your local and change webhookUrl by the generated one
+  // Consule README.md for more details
```
Comment (kimmy) [inline main.go]: what's `consule`?

File: main.go [hunk 9]
```diff
@@ -1,31 +1,79 @@
...
+
+  // webhookUrl := "https://{production_api_endpoint}/" + os.Getenv("TOKEN")
+	webhookUrl := "https://38e44fc0.ngrok.io/" + os.Getenv("TOKEN")
```
Comment (kimmy) [inline main.go]: should we have used an env variable?

File: models/telegram_update.go [hunk 10]
```diff
@@ -0,0 +1,14 @@
+package models
+
+import "github.com/jinzhu/gorm"
+
+// Change this to Telegram's update json
```
Comment (kimmy) [inline models/telegram_update.go]: Still necessary?

**Review by kimmy — CHANGES_REQUESTED** [review]
I think some changes are necessary. Stated in the comments.

File: .gitignore (hunk 1 above)
Comment (mlsaito) [reply .gitignore]: Sure thing.

File: main.go (hunk 9 above)
Comment (mlsaito) [reply main.go]: Got it.

File: main.go (hunk 8 above)
Comment (mlsaito) [reply main.go]: Typo. Will update to 'consult'.

File: controllers/controllers.go (hunk 3 above)
Comment (mlsaito) [reply controllers/controllers.go]: No. Bot logic (message parsing, bot reply) will be located here.

File: logger/logger.go (hunk 5 above)
Comment (mlsaito) [reply logger/logger.go]: Good point. Will update spacing from spaces to tabs.

**Review by kimmy — APPROVED** [review]

**Pull request merged**